acceptance/tests/wan-federation/wan_federation_test.go

```diff
@@ -423,20 +423,21 @@ func serviceFailoverCheck(t *testing.T, options *terratestK8s.KubectlOptions, po
 	var resp string
 	var err error
 
-	f := func(ft require.TestingT) {
+	// Retry until we get the response we expect, sometimes you get back the previous server until things stabalize
+	retry.RunWith(timer, t, func(r *retry.R) {
 		resp, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", "-i",
 			staticClientDeployment, "-c", connhelper.StaticClientName, "--", "curl", fmt.Sprintf("localhost:%s", port))
-		require.NoError(ft, err)
-		assert.Contains(ft, resp, expectedName)
-	}
+		assert.NoError(r, err)
+		assert.Contains(r, resp, expectedName)
+	})
 
+	// Try again to rule out load-balancing. Errors can still happen so retry
```

Comment:
~ I know this is status-quo so non-blocking: Do we know that LB would be round-robin in this scenario? Wondering whether multiple follow-up requests would be appropriate to prove we're not getting lucky. I did that in [this](https://github.com/hashicorp/consul-enterprise/blob/df2e31f83af2178aa75b53480a2f41f55390bb9d/test/integration/consul-container/test/basic/locality_aware_routing_ent_test.go#L130) test for similar reasons.

Reply:
good idea, added something similar